common/models/user.js

```diff
@@ -449,11 +449,16 @@ module.exports = function(User) {
       options.mailer || this.constructor.email || registry.getModelByType(loopback.Email);
 
     // Set a default token generation function if one is not provided
-    var tokenGenerator = options.generateVerificationToken || User.generateVerificationToken;
-
-    tokenGenerator(user, function(err, token) {
-      if (err) { return fn(err); }
+    let tokenGenerator = new Promise(function(resolve, reject){
```

Comment:
Instead of completely switching to `Promise` based implementation, I would suggest you add a check to test if tokenGenerator is a function to minimize the changes and keep it consistent.